src/plugins/intel_cpu/src/utils/serialize.hpp

```diff
@@ -41,6 +41,10 @@ class ModelDeserializer {
 
     void operator>>(std::shared_ptr<ov::Model>& model);
 
+    void set_weights_path(std::string& weights_path) {
+        m_weights_path = weights_path;
```

Comment:
please, note that case when model is compiled as `compile_model(ov::Model)` should also be supported (see PR https://github.com/openvinotoolkit/openvino/pull/29107, [NPU](https://github.com/openvinotoolkit/openvino/pull/29274) & GPU work is in progress) via `hint::model_ptr`